RInAppMessaging/Classes/UserDataCache.swift

```diff
@@ -0,0 +1,77 @@
+internal protocol UserDataCacheable: AnyObject {
+    func getUserData(identifiers: [UserIdentifier]) -> UserDataCacheContainer?
+    func cacheCampaignData(_ data: [Campaign], userIdentifiers: [UserIdentifier])
+    func cacheDisplayPermissionData(_ data: DisplayPermissionResponse, for campaign: Campaign, userIdentifiers: [UserIdentifier])
+}
+
+internal struct UserDataCacheContainer: Codable, Equatable {
+    fileprivate(set) var campaignData: [Campaign]?
+    fileprivate var displayPermissionData: [String: DisplayPermissionResponse]
+
+    init(campaignData: [Campaign]? = nil, displayPermissionData: [String: DisplayPermissionResponse] = [:]) {
+        self.campaignData = campaignData
+        self.displayPermissionData = displayPermissionData
+    }
+
+    func displayPermissionData(for campaign: Campaign) -> DisplayPermissionResponse? {
+        return displayPermissionData[campaign.id]
+    }
+}
+
+internal class UserDataCache: UserDataCacheable {
+
+    private let userDefaults: UserDefaults
+    private var cachedContainers: [Set<UserIdentifier>: UserDataCacheContainer]
```

Comment:
Would it be helpful to alias that type for conciseness?

Reply:
Yeah it might be 👍